pkg/api/types.go

```diff
@@ -180,6 +180,8 @@ type VolumeSource struct {
 	GitRepo *GitRepo `json:"gitRepo"`
 	// Secret represents a secret that should populate this volume.
 	Secret *SecretSource `json:"secret"`
+	// NFSMount represents an nfs mount on the host that shares a pod's lifetime
+	NFSMount *NFSMount `json:"nfsMount"`
```

Comment:
Can we just call this "NFS" for the variable and NFSVolumeSource for the type (PR in flight to rename all the rest to *VolumeSource")